src/lib/types.ts

```diff
@@ -347,6 +347,8 @@ export type SignInWithOAuthCredentials = {
     redirectTo?: string
     /** A space-separated list of scopes granted to the OAuth application. */
     scopes?: string
+    /** A custom OAuth callback URL. The proxy must redirect (with query params retained) to the Authorize end-point */
+    proxy?: string,
```

Comment:
As I voiced in the GoTrue PR, I believe there's a better name for `proxy`. Something along the lines of `redirect_via` or `redirect_through` probably conveys meaning better, but I'm open to any other name.